windows/sound.h

```diff
@@ -63,50 +61,62 @@ class CSound : public CSoundBase
-    virtual QString GetOutputChannelName ( const int iDiD ) { return channelInfosOutput[iDiD].name; }
-    virtual void    SetLeftOutputChannel  ( const int iNewChan );
-    virtual void    SetRightOutputChannel ( const int iNewChan );
-    virtual int     GetLeftOutputChannel()  { return vSelectedOutputChannels[0]; }
-    virtual int     GetRightOutputChannel() { return vSelectedOutputChannels[1]; }
-
-    virtual float   GetInOutLatencyMs() { return fInOutLatencyMs; }
-
-protected:
-    virtual QString  LoadAndInitializeDriver ( QString strDriverName,
-                                               bool    bOpenDriverSetup );
-    virtual void     UnloadCurrentDriver();
-    int              GetActualBufferSize ( const int iDesiredBufferSizeMono );
-    QString          CheckDeviceCapabilities();
-    bool             CheckSampleTypeSupported ( const ASIOSampleType SamType );
-    bool             CheckSampleTypeSupportedForCHMixing ( const ASIOSampleType SamType );
-    void             ResetChannelMapping();
-
-    int              iASIOBufferSizeMono;
-    int              iASIOBufferSizeStereo;
-
-    long             lNumInChan;
-    long             lNumInChanPlusAddChan; // includes additional "added" channels
-    long             lNumOutChan;
-    float            fInOutLatencyMs;
-    CVector<int>     vSelectedInputChannels;
-    CVector<int>     vSelectedOutputChannels;
+    virtual int GetNumInputChannels()
+    {
+        return static_cast<int> ( lNumInChanPlusAddChan );
+    }
+    virtual QString GetInputChannelName ( const int iDiD )
+    {
+        return channelInputName[iDiD];
+    }
+    virtual void SetLeftInputChannel ( const int iNewChan );
+    virtual void SetRightInputChannel ( const int iNewChan );
+    virtual int  GetLeftInputChannel() { return vSelectedInputChannels[0]; }
+    virtual int  GetRightInputChannel() { return vSelectedInputChannels[1]; }
+
+    virtual int GetNumOutputChannels()
+    {
+        return static_cast<int> ( lNumOutChan );
+    }
+    virtual QString GetOutputChannelName ( const int iDiD )
+    {
+        return channelInfosOutput[iDiD].name;
+    }
+    virtual void SetLeftOutputChannel ( const int iNewChan );
+    virtual void SetRightOutputChannel ( const int iNewChan );
+    virtual int  GetLeftOutputChannel() { return vSelectedOutputChannels[0]; }
+    virtual int  GetRightOutputChannel() { return vSelectedOutputChannels[1]; }
+
+    virtual float GetInOutLatencyMs() { return fInOutLatencyMs; }
+
+  protected:
+    virtual QString LoadAndInitializeDriver ( QString strDriverName,
+                                              bool    bOpenDriverSetup );
+    virtual void    UnloadCurrentDriver();
+    int             GetActualBufferSize ( const int iDesiredBufferSizeMono );
+    QString         CheckDeviceCapabilities();
+    bool            CheckSampleTypeSupported ( const ASIOSampleType SamType );
```

Comment:
Why does it suddenly drop line 99 back to the left, rather than continue alignment?  That's strange...

Reply:
that's because of the `ColumnLimit` which is 80. The linter is configured so staying below that limit is its highest priority by having configured the highest `Penalty` on `PenaltyExcessCharacter`:
```
PenaltyBreakAssignment: 2
PenaltyBreakBeforeFirstCallParameter: 19
PenaltyBreakComment: 300
PenaltyBreakFirstLessLess: 120
PenaltyBreakString: 1000
PenaltyBreakTemplateDeclaration: 10
PenaltyExcessCharacter: 1000000
PenaltyReturnTypeOnItsOwnLine: 60
```

So that could be changed in the following ways if preferred:
- have a bigger limit (which gets enforced)
- tune penalties so the limit may be exceeded in favor of code alignments

Reply:
> tune penalties so the limit may be exceeded in favor of code alignments

I guess this would be worth a try as this would most likely resemble what human code editors have done in the past: Use a character limit, but extend it if it helps readability in other ways.

Reply:
I have now tried few different variants with the `Penalty` settings but didn't really get better results with that. Alignments like in the example mentioned seem to be always last priority and are only done when the column limit allows for it.

So only alternative would be to choose a bigger limit, resulting in more alignments that fit within the limit.